### ISSUE TITLE
Taint Analysis: hoisting a `where()` array literal into a variable reinstates the bound-value false positive

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1307]
Surfaced by an external review of #1305. This is the residual false positive #1302 explicitly left in place, now with a concrete repro.

#1302 made `where()` array **literals** strip element-wise, so the PDO-bound positions no longer carry a SQL sink. Hoisting the identical array into a variable falls back to the old type-level path and the false positive returns:

```php
// silent (correct) — argument is an Array_ literal, element-wise strip applies
$query->where([['name', 'LIKE', '%' . (string) $request->term . '%']]);

// TaintedSql (false positive) — same value, same bound position
$conditions = [['name', 'LIKE', '%' . (string) $request->term . '%']];
$query->where($conditions);
```

Ordinal 2 of a nested condition is passed to `addBinding()` and is never emitted as SQL, so neither form is injectable.

Mechanism: element-wise recording only runs when the call argument itself is an `Array_`:

```php
if ($argument instanceof Array_ && !$expr instanceof StaticCall) {   // WhereColumnTaintHandler.php:230
    self::recordBoundValuePositions($argument, $expr->var);
}
```

A hoisted variable therefore reaches `isBoundValueMap()`, which rejects the numeric outer key (:527) and retains the sink.

This is a documented limitation rather than a regression — #1302's description states that "a non-literal argument (`$conds = [...]; where($conds)`) still takes the old type-level path unchanged". Filing it so the residual FP is tracked rather than folklore.

### Scope

- Decide whether hoisted-literal support is worth the machinery: the value's type is a sealed `TKeyedArray` at the call site, so the bound positions are recoverable from the **type** without needing the `ArrayItem` identity the current strip keys on. That is a different mechanism from the literal walk, not an extension of it.
- Cheaper alternative worth pricing first: assignment-provenance (resolve the variable to its single defining `Array_` literal when unambiguous) and reuse the existing walk. Prior art for provenance restriction: #1070 / PR #1073.
- Either way it needs the same guards as #1306, and phpt coverage mirroring `SafeSqlWhereNestedConditionValues.phpt` in hoisted form.

**Comment by alies-dev** [comment]
Closing as won't-fix on ROI grounds, after measuring prevalence rather than assuming it.

Grepped the 18-app benchmark corpus (bagisto, coolify, monica, pixelfed, filament, bookstack, aureuserp, unit3d, vito, tastyigniter, solidtime, corcel, laravel-excel, laravel-backup, laravel-permission, laravel-socialite, spatie-dashboard, and one private app) for the hoisted form:

```
->where($someVariable)
```

**Zero occurrences.** The false positive described here needs an app to hoist a nested-condition array literal into a variable before passing it to `where()`, and no app in the corpus does that.

Against that, the fix is not cheap: the bound positions would have to be recovered from the argument's *type* (a sealed `TKeyedArray`) rather than from `ArrayItem` identity, which is a different mechanism from the literal walk, not an extension of it. High cost, no demonstrated incidence.

The limitation is already documented in #1302's description ("a non-literal argument still takes the old type-level path unchanged"), so closing this loses no knowledge — it only removes duplicate tracking.

Reopen if a real codebase turns up the shape.
